riscv/csrs.h

```diff
@@ -787,4 +791,37 @@ class jvt_csr_t: public basic_csr_t {
   jvt_csr_t(processor_t* const proc, const reg_t addr, const reg_t init);
   virtual void verify_permissions(insn_t insn, bool write) const override;
 };
+
+class sscsrind_sel_csr_t : public basic_csr_t {
+ public:
+  typedef std::shared_ptr<sscsrind_sel_csr_t> sscsrind_sel_csr_t_p;
+  sscsrind_sel_csr_t(processor_t* const proc, const reg_t addr, const reg_t init,
+                     sscsrind_sel_csr_t_p virt_csr = nullptr);
+  reg_t read() const noexcept override;
+  virtual void verify_permissions(insn_t insn, bool write) const override;
+ protected:
+  virtual bool unlogged_write(const reg_t val) noexcept override;
+ private:
+  sscsrind_sel_csr_t_p virt_csr;
```

Comment:
These classes appear to be reimplementing `virtualized_csr_t`. Is there any reason these classes can't be simple nonvirtualized CSRs, then instantiated using `virtualized_csr_t`? See `stval` for a typical example:

```
  nonvirtual_stval = std::make_shared<basic_csr_t>(proc, CSR_STVAL, 0);
  csrmap[CSR_VSTVAL] = vstval = std::make_shared<basic_csr_t>(proc, CSR_VSTVAL, 0);
  csrmap[CSR_STVAL] = stval = std::make_shared<virtualized_csr_t>(proc, nonvirtual_stval, vstval);
```

Reply:
One of the limitations of `virtualized_csr_t` is that doesn't override `verify_permissions()`. That said, the code has undergone significant changes; will post an update.

Reply:
Another alternative is to add `verify_permissions()` to `virtualized_csr_t`, but didn't want to risk introducing unintended side-effects.